### ISSUE TITLE
Traefik on k8s not listening externally without changing deployment

### ISSUE THREAD

**Issue by kallisti5** [containous/traefik#3951]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Attempt to deploy traefik as an ingress controller on a single node kubernetes cluster.

### What did you expect to see?

Traefik listing on 80/8080

### What did you see instead?

Traefik DS was only listening internally and not externally.

### Output of `traefik version`: (_What version of Traefik are you using?_)

1.7.0

### Reproduction Steps

Kubernetes v1.11.3

* Fedora 28
* Disable selinux
* apply updates
* ```kubeadm init --pod-network-cidr=10.244.0.0/16 -v1```
* ```sysctl net.bridge.bridge-nf-call-iptables=1``` (and put int sysctl.conf)
* ```kubectl apply -f https://raw.githubusercontent.com/coreos/flannel/v0.10.0/Documentation/kube-flannel.yml```
* ```kubectl apply -f https://raw.githubusercontent.com/containous/traefik/master/examples/k8s/traefik-rbac.yaml```
* ```kubectl apply -f https://raw.githubusercontent.com/containous/traefik/master/examples/k8s/traefik-ds.yaml```

A dump of ```kubectl --namespace=kube-system get pods``` shows traefik running on an internal IP, but not listening externally.

### Workaround

Adding ```hostNetwork: true``` to the traefik-ds.yaml seemed to solve the issue and open traefik to the internet.

```patch
$ diff -Naur traefik-ds-original.yaml traefik-ds.yaml 
--- traefik-ds-original.yaml	2018-09-28 20:37:21.124852368 +0000
+++ traefik-ds.yaml	2018-09-28 20:37:53.963960171 +0000
@@ -21,6 +21,7 @@
     spec:
       serviceAccountName: traefik-ingress-controller
       terminationGracePeriodSeconds: 60
+      hostNetwork: true
       containers:
       - image: traefik
         name: traefik-ingress-lb
```

**Comment by kallisti5** [comment]
before change:
```
$ sudo netstat -an | grep LISTEN
tcp        0      0 127.0.0.1:10249         0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:10251         0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:2379          0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:10252         0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:2380          0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:46221         0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:10248         0.0.0.0:*               LISTEN     
tcp6       0      0 :::10250                :::*                    LISTEN     
tcp6       0      0 :::6443                 :::*                    LISTEN     
tcp6       0      0 :::10256                :::*                    LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
tcp6       0      0 :::9090                 :::*                    LISTEN     
```

After Change:
```
$ netstat -an | grep LISTEN
tcp        0      0 127.0.0.1:10249         0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:10251         0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:2379          0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:10252         0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:2380          0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:46221         0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:10248         0.0.0.0:*               LISTEN     
tcp6       0      0 :::10250                :::*                    LISTEN     
tcp6       0      0 :::6443                 :::*                    LISTEN     
tcp6       0      0 :::80                   :::*                    LISTEN     
tcp6       0      0 :::8080                 :::*                    LISTEN     
tcp6       0      0 :::10256                :::*                    LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
tcp6       0      0 :::9090                 :::*                    LISTEN     
```

**Comment by traefiker** [comment]
Hi! I'm Træfiker :robot: the bot in charge of communication regulation.

Thanks for your interest in Træfik!

We dedicate the issue tracker to bug reports and feature requests only.

If you have questions and need community support, you can join our [Slack workspace](https://slack.traefik.io/) and reach out to us on the [#support channel](https://traefik.slack.com/messages/C0CDT22PJ/).

Another option is to use [Stack Overflow](https://stackoverflow.com/questions/tagged/traefik) and tag your question `traefik`.

Of course, double checking [Træfik's documentation](https://docs.traefik.io/) is always a good idea :sweat_smile:

**Comment by kallisti5** [comment]
Isn't this an issue in the documentation?

**Comment by alekc** [comment]
@kallisti5 Thx for this issue, it was driving me crazy that I couldn't deploy traefik by following official guide...

I agree, this should be classified as bug in documentaiton.

**Comment by alekc** [comment]
In my case cluster was deployed via kubeadm on baremetal (so no loadbalancer to give ip to kubernetes), which might've been the cause. 

hostNetwork clause was present in daemonset configuration on traefik 1.5 (https://docs.traefik.io/v1.5/user-guide/kubernetes/), it have been removed for some reason. 

This https://github.com/containous/traefik/commit/10ca35dccdeb02b8d6baa6ea8816bc3a727ae3ec commit has removed. ...

Anyway after your comment, I tried to wipe everything again and it worked this time... beat me...
